store.go

```diff
@@ -39,6 +41,37 @@ func (c *controller) initDataStore() error {
 	return c.watchNetworks()
 }
 
+func (c *controller) initLocalDataStore() error {
+	c.Lock()
+	cfg := c.cfg
+	c.Unlock()
+	// init local datastore
+	if !c.validateLocalDatastoreConfig() {
+		return fmt.Errorf("local datastore initialization requires a valid configuration")
```

Comment:
I dont think we should fail if the user hasnt configured a local store.
Since boltdb is an embedded db (without the need for any external process), we should use that as the default and have a predefined location to store the local database. The user can always override that default setting through the configuration.